src/lib/bootloader/proposal_client.rb

```diff
@@ -172,28 +173,48 @@ def construct_proposal_map
     # Add to argument proposal map all errors detected by proposal
     # @return modified parameter
     def handle_errors(ret)
-      if ::Bootloader::BootloaderFactory.current.name == "none"
+      current_bl = ::Bootloader::BootloaderFactory.current
+      if current_bl.name == "none"
         log.error "No bootloader selected"
         ret["warning_level"] = :warning
         # warning text in the summary richtext
         ret["warning"] = _(
           "No boot loader is selected for installation. Your system might not be bootable."
         )
+        return
       end
 
       if !Yast::BootStorage.bootloader_installable?
         ret["warning_level"] = :error
         ret["warning"] = _(
           "Because of the partitioning, the bootloader cannot be installed properly"
         )
+        return
       end
 
       if !Yast::BootSupportCheck.SystemSupported
         ret["warning_level"] = :error
         ret["warning"] = Yast::BootSupportCheck.StringProblems
+        return
       end
 
-      ret
+      pkgs = current_bl.packages.map { |p| [p, Yast::Pkg.ResolvableProperties(p, :package, "")] }
+      log.info "packages info #{pkgs.inspect}"
+      pkgs.select! { |_n, p| unselected?(p) }
+      return if pkgs.empty?
+
+      ret["warning_level"] = :error
+      ret["warning"] = _("Packages required for booting is deselected (%s). " \
```

Comment:
If you look at the screenshot the wording seems a bit weird. "Packages required ... (grub2)" - there is  *Packages* but there is only *one* package showed.

Better use the plural form:
```ruby
# TRANSLATORS: %s are package names
n_("A package required for booting is deselected (%s). Please select it for installation again.",
  "Packages required for booting are deselected (%s). Please select them for installation again.",
  pkgs.size) % pkgs.map(&:first).join(", ")
```

This will look much better.